File: consultorio-frontend/src/app/models/cita.model.ts
export type EstadoCita = 'Programada' | 'Confirmada' | 'EnCurso' | 'Completada';

export interface Cita {
  citaId: number;
  pacienteId: number;
  pacienteNombre: string;
  fecha: string; // ISO date (yyyy-MM-dd)
  horaInicio: string; // HH:mm
  horaFin: string; // HH:mm
  estado: EstadoCita;
  motivo?: string;
  notas?: string;
}

export interface AgendaSemanalDia {
  fecha: string;
  dia: string;
  citas: Cita[];
}

export interface DailyStats {
  fecha: string;
  programadas: number;
  confirmadas: number;
  enCurso: number;
  completadas: number;
  total: number;
  horasOcupadas: number;
  horasDisponibles: number;
}

export interface CitaPayload {
  pacienteId: number;
  fecha: string; // yyyy-MM-dd
  horaInicio: string; // HH:mm
  horaFin: string; // HH:mm
  motivo?: string;
  notas?: string;
}

File: consultorio-backend/Models/CitaModel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ConsultorioMedico.API.Models
{
    public class CitaModel
    {
        [Key]
        public int CitaId { get; set; }

        [Required]
        public int PacienteId { get; set; }

        public PacienteModel? Paciente { get; set; }

        [Required]
        public DateTime Fecha { get; set; }

        [Required]
        public TimeSpan HoraInicio { get; set; }

        [Required]
        public TimeSpan HoraFin { get; set; }

        [Required]
        public EstadoCita Estado { get; set; } = EstadoCita.Programada;

        [StringLength(500)]
        public string? Motivo { get; set; }

        [StringLength(1000)]
        public string? Notas { get; set; }

        public DateTime FechaCreacion { get; set; } = DateTime.Now;
    }

    public enum EstadoCita
    {
        Programada,
        Confirmada,
        EnCurso,
        Completada
    }
}
